modApiServer/src/org/aion/api/server/ApiAion.java

```diff
@@ -535,47 +533,49 @@ public BigInteger getNonce(Address _address) {
         return this.ac.getRepository().getNonce(_address);
     }
 
-    protected byte[] sendTransaction(ArgTxCall _params) {
+    protected ApiTxResponse sendTransaction(ArgTxCall _params) {
+
+        if (_params == null)
+            return(new ApiTxResponse(AddTxResponse.INVALID_TX));
 
         Address from = _params.getFrom();
 
         if (from == null || from.isEmptyAddress()) {
             LOG.error("<send-transaction msg=invalid-from-address>");
-            return null;
+            return(new ApiTxResponse(AddTxResponse.INVALID_FROM));
         }
 
         ECKey key = this.getAccountKey(from.toString());
         if (key == null) {
             LOG.error("<send-transaction msg=account-not-found>");
-            return null;
+            return(new ApiTxResponse(AddTxResponse.INVALID_ACCOUNT));
         }
 
         try {
             synchronized (pendingState) {
                 // TODO : temp set nrg & price to 1
                 byte[] nonce =
-                        (!_params.getNonce().equals(BigInteger.ZERO))
-                                ? _params.getNonce().toByteArray()
-                                : pendingState
-                                        .bestPendingStateNonce(Address.wrap(key.getAddress()))
-                                        .toByteArray();
+                    (!_params.getNonce().equals(BigInteger.ZERO))
+                        ? _params.getNonce().toByteArray()
+                        : pendingState
+                            .bestPendingStateNonce(Address.wrap(key.getAddress()))
+                            .toByteArray();
 
                 AionTransaction tx =
-                        new AionTransaction(
-                                nonce,
-                                _params.getTo(),
-                                _params.getValue().toByteArray(),
-                                _params.getData(),
-                                _params.getNrg(),
-                                _params.getNrgPrice());
+                    new AionTransaction(
+                        nonce,
+                        _params.getTo(),
+                        _params.getValue().toByteArray(),
+                        _params.getData(),
+                        _params.getNrg(),
+                        _params.getNrgPrice());
                 tx.sign(key);
 
-                pendingState.addPendingTransaction(tx);
-
-                return tx.getHash();
+                return (new ApiTxResponse(pendingState.addPendingTransaction(tx), tx.getHash()));
             }
         } catch (Exception ex) {
-            return null;
+            LOG.error("ApiAion.sendTransaction exception: [{}]", ex.getMessage());
+            return(new ApiTxResponse(AddTxResponse.EXCEPTION, ex));
         }
     }
 
```

Comment:
missing implement in the following method

Reply:
done